### ISSUE TITLE
Add GUI for editing dub.json/dub.sdl

### ISSUE THREAD

**Issue by WebFreak001** [Pure-D/code-d#75]
Like markdown preview or installing extensions

Draft:
General:
![General](https://i.webfreak.org/fLtcbG)
Build Settings:
![Build Settings](https://i.webfreak.org/fLtJoG)
Sources:
![Sources](https://i.webfreak.org/fLt9Ss)
Dependencies:
![Dependencies](https://i.webfreak.org/fLtnK0)
Build Types:
![Build Types](https://i.webfreak.org/fLLY38)
Configurations:
![Configurations](https://i.webfreak.org/fLLsaH)

Any feedback? (@Vild @yannick and other users)

**Comment by ZombineDev** [comment]
Very nice! Another killer features would be auto-completion for the dependencies. You can use the official API to get a list of available packages:
http://code.dlang.org/api/packages/search

**Comment by ZombineDev** [comment]
Small nitpick about the UI - on some of the views (e.g. General) it may not be 100% clear the correspondence between fields and their titles. May be you can put each title + field in a rounded rectangle with a little bit darker background to highlight the grouping.

**Comment by ZombineDev** [comment]
Also, since this is targeted at "code-d for noobs" :smile: you may want to hide all those checkboxes in build settings under a "More options" button, in order not to scare the new users :D

**Comment by ZombineDev** [comment]
BTW, make sure to use colors from the currently set theme. It's often annoying when plugins for other programs have inconsistent UI with the rest of the application.

**Comment by WebFreak001** [comment]
ok yeah thanks for the feedback. Gonna take the theme colors into account and the labels/input boxes will probably look different anyway. That was just a drawing I made in inkscape and I couldn't really get an input box to look nice, so I took a simple material design approach and didn't care about the spacing when I made it. Maybe I will put the build requirements and build options checkboxes in some separate tab too.
